apps/client/src/module/dashboard/profile/profile-service.ts

```diff
@@ -0,0 +1,65 @@
+"use server";
+
+import {
+  getServerGqlClient,
+  graphql,
+} from "@src/module/graphql";
+import { TypedDocumentNode } from '@graphql-typed-document-node/core';
+
+interface UpdateUserResult {
+  updateUser: boolean;
+}
+
+interface UpdateUserVariables {
+  id: string;
+  input: {
+    displayName: string;
+  };
+}
+
+const updateUserMutation = graphql(`
+    mutation UpdateUserDisplayName($id: String!, $input: UpdateUserInput!) {
+        updateUser(id: $id, input: $input)
+    }
+`) as TypedDocumentNode<UpdateUserResult, UpdateUserVariables>;
+
+export type UpdateUserPayload = {
+  id: string;
+  displayName: string;
+};
+
+function validateUpdatedDisplayName(name: string): boolean {
+    const regex = /^[a-zA-Z0-9-_]+$/;
```

Comment:
are we still using regex on form validation, not zod?

Reply:
this validation part should be made on the form, not here

Reply:
fixed